src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = [];
+  let newStateItem = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(newStateItem, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete newStateItem[key];
+        }
+        break;
+
+      case 'clear':
+        newStateItem = {};
+        break;
+
+      default:
+        return newStateItem;
```

Comment:
if you look at the description of your method, you'll see return value type `Object[]`, thus here you shouldn't return anything